lib/useData.js

```diff
@@ -151,7 +144,24 @@ export const DataProvider = ({ children }) => {
     if (response.ok) mutateEvent();
   }
 
-  // Function adding a comment to an event and mutate event
+  async function updateUser(event, user) {
+    const data = new FormData(event.target);
+    const formData = Object.fromEntries(data);
+    const userData = { ...user, ...formData };
+    const response = await fetch(`/api/user/${userData.id}`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(userData),
+    });
+
+    if (response.ok) {
+      return true;
+    } else if (!response.ok) {
+      return false;
+    }
+  }
```

Comment:
- Ensure the default form submission behaviour is prevented

```suggestion
  async function updateUser(event, user) {
   event.preventDefault();
    const data = new FormData(event.target);
    const formData = Object.fromEntries(data);
    const userData = { ...user, ...formData };
    const response = await fetch(`/api/user/${userData.id}`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(userData),
    });

    if (response.ok) {
      return true;
    } else if (!response.ok) {
      return false;
    }
  }
```